docs/madr/decisions/014-mesh-circuit-breaker-policy.md

```diff
@@ -0,0 +1,289 @@
+debugging process, may look only at stats prefixed with `cluster.<name>.circuit_breakers.`, but according to the
+behaviour of our policy, if the data plane was ejected because of one of the faults defined in `detectors` section,
+stats which contain interesting us data will be prefixed with `cluster.<name>.outlier_detection.`.
+
+## Considered Options
+
+* Create two separate policies: `MeshCircuitBreaker` and `MeshOutlierDetector`
+* Create single `MeshCircuitBreaker` policy which would map 1:1 current policy
+* Create single `MeshCircuitBreaker` with configuration divided
+  to `connectionThresholds` ([envoy's circuit breaker](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/circuit_breaking))
+  and `outlierDetection` ([envoy's outlier detection](https://gstwww.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/outlier))
+
+## Decision Outcome
+
+Chosen option: Create single `MeshCircuitBreaker` with configuration divided into two sections.
+
+## Positive Consequences
+
+* Having two separate policies, each with explicit configurations makes it easier to look for appropriate Envoy stats
+  and XDP configuration values.
+* By combining two abstractions, which exist in Envoy, and calling them both combined with a name of one of these is
+  confusing, especially when you are trying to debug some issue without deep understanding of internals of our service
+  mesh.
+* Default `MeshCircuitBreaker` policy is sufficient to replicate behaviour of current version of this policy - there
+  won't be a need to include a default `MeshOutlierDetector` policy
+
+## Negative Consequences
+
+* `MeshCircuitBreaker` will be a policy with very small amount of configuration available, which may introduce some
+  clutter (at some point there may be so many policies, that it will be much more difficult to understand all of them)
+
+## Solution
+
+By dividing the configuration section into two separate, logical sections: `connectionThresholds` and `outlierDetection`
+it's easier to differentiate underlying envoy features without the need to split policy into two separate ones.
+
+Introducing new - 2.0 policy is a great opportunity to introduce missing so far functionality of circuit breakers for
+inbound clusters, and with new policy matching it may be so simple to just allow to specify`from.targetRef` section
+
+Additional changes to the original `CircuitBreaker` policy consists of:
+
+- changing word `Errors` to `Failures` in detectors and adding `origin` suffix to the local failures
+  detector (`totalErrors`, `gatewayErrors` and `localErrors`becomes: `totalFailures`, `gatewayFailures`
+  and `localOriginFailures` respectively)
+- adding additional threshold-related property - `maxConnectionPools` for specifying the maximum number of connection
+  pools per cluster that Envoy will concurrently support at once
+
+### `connectionThresholds`
```

Comment:
I'm still on a fence with this name. I'm also thinking on just simple `thresholds`, but I'm not sure. Wdyt?

Reply:
connectionBreakers or connectionLimits otherwise?

Reply:
🤔

Reply:
I like `connectionLimits`